### PR TITLE
feat: magic of await (#40014)

### PR DESCRIPTION
**Pull request by nx-mukesh**
User login API added

### CODE REVIEW

File: src/controller/user/Controller.ts [hunk 1]
```diff
@@ -3,168 +3,196 @@ import UserRepository from '../../repositories/user/UserRepository';
...
+			if (!user) {
+				return next({ status: 404, error: 'bad request', message: 'User Not found' });
+			}
+			const validPassword = bcrypt.compareSync(password, user.password);
+      // console.log({validPassword});
+      
+			if (!validPassword) {
+				return next({ success: false, message: 'Password not matched' });
+			}
+			const token = await jwt.sign(req.body, config.secret, { expiresIn: '15m' });
```
Comment (namanpar) [inline src/controller/user/Controller.ts]: req.body would not sent.
Either user or some other sepcific value would be sent in token

File: src/repositories/versionable/VersionableRepository.ts [hunk 2]
```diff
@@ -6,116 +6,104 @@ import { idText } from 'typescript';
...
+	 * @param projection
+	 * @param options
+	 * @returns Array of Documents
+	 */
+	protected find(query, projection?: any, options?: any): Query<EnforceDocument<I, {}>[], EnforceDocument<I, {}>> {
+		const { skip = 0, limit = 10, sortBy = '-createdAt' } = query;
+		const finalQuery = { deletedAt: undefined, ...query };
+		const findData = this.model.find(finalQuery, projection, options);
+		console.log(findData);
+		return this.model.find(finalQuery, projection, { skip: +skip, limit: +limit }).sort(`-${sortBy}`);
```
Comment (namanpar) [inline src/repositories/versionable/VersionableRepository.ts]: what is this +skip and +limit

File: src/controller/user/routes.ts [hunk 3]
```diff
@@ -19,12 +19,24 @@ router.get(
...
   validationHandler(validation.create),
   UserController.create
 );
+router.post(
+  '/login',
+  // authMiddleware(USER, 'read'),
+  validationHandler(validation.create),
+  UserController.login
+);
 router.put(
```
Comment (namanpar) [inline src/controller/user/routes.ts]: login route is defined 2 times

**Review by namanpar — CHANGES_REQUESTED** [review]

File: src/repositories/versionable/VersionableRepository.ts (hunk 2 above)
Comment (nx-mukesh) [reply src/repositories/versionable/VersionableRepository.ts]: +skip and +limit is query to find limited document with search query.
Like localhost:900/api/user?skip=0&limit=2&search=john => return 2 document with match of john

File: src/controller/user/Controller.ts (hunk 1 above)
Comment (nx-mukesh) [reply src/controller/user/Controller.ts]: fixed as per feedback

File: src/controller/user/routes.ts (hunk 3 above)
Comment (nx-mukesh) [reply src/controller/user/routes.ts]: fixed

File: src/repositories/versionable/VersionableRepository.ts (hunk 2 above)
Comment (namanpar) [reply src/repositories/versionable/VersionableRepository.ts]: What does + signify here?

Comment (nx-mukesh) [reply src/repositories/versionable/VersionableRepository.ts]: its doing nothing just a showcase of should be positive and number only.

**Review by namanpar — APPROVED** [review]

**Pull request merged**